### DISCUSSION TITLE
Version 7.44.0-rc.4

### DISCUSSION THREAD

**Discussion by bluebill1049** [react-hook-form/react-hook-form#10427] [General]
👔 Improve Form component `onSubmit` and remove `fetcher` prop (#10425)

```diff
<Form 
-  fetcher={(...) => {}}
+  onSubmit={async ({ formData, }) => {
+    await fetch('/api', { body: formData, method: 'post' })
+  }} 
/>
```

```tsx
// With action prop, it achieve the same as above
<Form action="/api" />
```

😵‍💫 close #10386 re-register controller input to fix strict mode (#10418)
✍️ update form.tsx for TSdoc (#10399)

<hr /><em>This discussion was created from the release <a href='https://github.com/react-hook-form/react-hook-form/releases/tag/v7.44.0-rc.4'>Version 7.44.0-rc.4</a>.</em>

**Comment by CanRau** [comment]
Believe there's a typo `onSubmit={await` should probably be `onSubmit={async` right?

**Reply by bluebill1049** [reply]
oh yea, let me fix it.

**Reply by bluebill1049** [reply]
Which line? couldn't find it.

**Reply by Moshyfawn** [reply]
> Which line? couldn't find it.

I've already edited it 😋 Sry, I thought you'd see the "edited by" label next to it

**Reply by bluebill1049** [reply]
lol thanks @Moshyfawn 🙏
